src/java/org/apache/cassandra/db/marshal/IntegerType.java

```diff
@@ -131,6 +143,336 @@ public static <VL, VR> int compareIntegers(VL lhs, ValueAccessor<VL> accessorL,
         return 0;
     }
 
+    /**
+     * Constructs a byte-comparable representation of the number.
+     *
+     * In the current format we represent it:
+     *    directly as varint, if the length is 6 or smaller (the encoding has non-00/FF first byte)
+     *    <signbyte><length as unsigned integer - 7><7 or more bytes>, otherwise
+     * where <signbyte> is 00 for negative numbers and FF for positive ones, and the length's bytes are inverted if
+     * the number is negative (so that longer length sorts smaller).
+     *
+     * Because we present the sign separately, we don't need to include 0x00 prefix for positive integers whose first
+     * byte is >= 0x80 or 0xFF prefix for negative integers whose first byte is < 0x80. Note that we do this before
+     * taking the length for the purposes of choosing between varint and full-form encoding.
+     *
+     * The representations are prefix-free, because the choice between varint and full-form encoding is determined by
+     * the first byte where varints are properly ordered between full-form negative and full-form positive, varint
+     * encoding is prefix-free, and full-form representations of different length always have length bytes that differ.
+     *
+     * Examples:
+     *    -1            as 7F
+     *    0             as 80
+     *    1             as 81
+     *    127           as C07F
+     *    255           as 80FF
+     *    2^32-1        as F8FFFFFFFF
+     *    2^32          as F900000000
+     *    2^56-1        as FEFFFFFFFFFFFFFF
+     *    2^56          as FF000100000000000000
```

Comment:
nit: Just a general note around the integer encoding docs...it might be helpful if we used a standard set of examples. It might make it a little easier for readers to jump between them. (ex. 2^32 is missing for `VariableLengthInteger` but present in both formats here.)